adapters/handlers/rest/handlers_objects.go

```diff
@@ -186,100 +191,110 @@ func (h *objectHandlers) getObjects(params objects.ObjectsListParams,
 	if err != nil {
 		switch err.(type) {
 		case errors.Forbidden:
-			return objects.NewObjectsUpdateForbidden().
-				WithPayload(errPayloadFromSingleErr(err))
-		case usecasesObjects.ErrInvalidUserInput:
-			return objects.NewObjectsUpdateUnprocessableEntity().
+			return objects.NewObjectsClassDeleteForbidden().
 				WithPayload(errPayloadFromSingleErr(err))
+		case uco.ErrNotFound:
+			return objects.NewObjectsClassDeleteNotFound()
 		default:
-			return objects.NewObjectsUpdateInternalServerError().
+			return objects.NewObjectsClassDeleteInternalServerError().
 				WithPayload(errPayloadFromSingleErr(err))
 		}
 	}
 
-	propertiesMap, ok := object.Properties.(map[string]interface{})
-	if ok {
-		object.Properties = h.extendPropertiesWithAPILinks(propertiesMap)
-	}
-
-	return objects.NewObjectsUpdateOK().WithPayload(object)
+	return objects.NewObjectsClassDeleteNoContent()
 }
 
-func (h *objectHandlers) deleteObject(params objects.ObjectsDeleteParams,
-	principal *models.Principal) middleware.Responder {
-	err := h.manager.DeleteObject(params.HTTPRequest.Context(), principal, params.ID)
+func (h *objectHandlers) updateObject(params objects.ObjectsClassPutParams,
+	principal *models.Principal,
+) middleware.Responder {
+	object, err := h.manager.UpdateObject(params.HTTPRequest.Context(), principal, params.ClassName, params.ID, params.Body)
 	if err != nil {
 		switch err.(type) {
 		case errors.Forbidden:
-			return objects.NewObjectsDeleteForbidden().
+			return objects.NewObjectsClassPutForbidden().
 				WithPayload(errPayloadFromSingleErr(err))
-		case usecasesObjects.ErrNotFound:
-			return objects.NewObjectsDeleteNotFound()
+		case uco.ErrInvalidUserInput:
+			return objects.NewObjectsClassPutUnprocessableEntity().
+				WithPayload(errPayloadFromSingleErr(err))
+		case uco.ErrNotFound:
+			return objects.NewObjectsClassDeleteNotFound()
 		default:
-			return objects.NewObjectsDeleteInternalServerError().
+			return objects.NewObjectsClassPutInternalServerError().
 				WithPayload(errPayloadFromSingleErr(err))
 		}
 	}
 
-	return objects.NewObjectsDeleteNoContent()
+	propertiesMap, ok := object.Properties.(map[string]interface{})
+	if ok {
+		object.Properties = h.extendPropertiesWithAPILinks(propertiesMap)
+	}
+
+	return objects.NewObjectsClassPutOK().WithPayload(object)
 }
 
-func (h *objectHandlers) headObject(params objects.ObjectsHeadParams,
-	principal *models.Principal) middleware.Responder {
-	exists, err := h.manager.HeadObject(params.HTTPRequest.Context(), principal, params.ID)
+func (h *objectHandlers) headObject(r objects.ObjectsClassHeadParams,
+	principal *models.Principal,
+) middleware.Responder {
+	ok, err := h.manager.HeadObject(r.HTTPRequest.Context(), principal, r.ClassName, r.ID)
 	if err != nil {
-		switch err.(type) {
-		case errors.Forbidden:
-			return objects.NewObjectsHeadForbidden().
+		switch {
+		case ok && err.Forbidden():
+			return objects.NewObjectsClassHeadForbidden().
```

Comment:
I think that this case should not have `ok` return to be tested, bc right now user will get a `ClassHeadForbidden` only if the object exists in the DB, but if the object doesn't exist he will get a proper `HeadNotFound` response. I think that `ClassHeadForbidden` has to be returned both for object `found` and `not found` cases to be consistent in the responses in case  that user is `Forbidden` to perform those operations

Reply:
Good spot!👍